src/App.tsx

```diff
@@ -0,0 +1,62 @@
+import { Component } from 'react';
+import './App.scss';
+import Header from './components/Header.tsx';
+import ResultsTable from './components/ResultsTable.tsx';
+import type { Pokemon } from './services/api.service.tsx';
+import { BounceLoader } from 'react-spinners';
+
+interface AppState {
+  pokemons: Pokemon[];
+  isLoading: boolean;
+}
+
+class App extends Component<{}, AppState> {
+  constructor(props: AppState) {
+    super(props);
+    this.state = {
+      pokemons: [],
+      isLoading: false,
+    };
+  }
+
+  setPokemons = (data: Pokemon[]) => {
+    this.setState({ pokemons: data });
+  };
+
+  setIsLoading = (loading: boolean) => {
+    this.setState({ isLoading: loading });
+  };
+
+  triggerFakeError = () => {
+    window.dispatchEvent(
+      new CustomEvent('GlobalError', {
+        detail: { message: 'This is a test error triggered manually.' },
+      })
+    );
+  };
+
+  render() {
+    const { pokemons, isLoading } = this.state;
+
```

Comment:
add  inside `render`:
```suggestion
 if (this.state.fakeerror) {
      throw new Error(`This is a test error triggered manually.`);
 }
```